alembic/script/revision.py

```diff
@@ -771,6 +771,15 @@ def iterate_revisions(
 
         """
 
+        if select_for_downgrade:
```

Comment:
**mike bayer** (zzzeek) wrote:

I think _iterate_revisions() might be better off split up into an _iterate_downgrade_revisions() / /_iterate_upgrade_revisions() for now as it seems to be split among a big conditional.

Otherwise this call to _iterate_revisions looks the same as the one at the bottom and the "select_for_downgrade" conditional can be reversed so that we only have to state _iterate_revisions once.   But I think for the moment splitting _iterate_revisions into two would be better.

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/2530

Reply:
**Simon Bowly** (simonbowly) wrote:

Done - realised while doing this that I was ignoring some options.  Can you clarify behaviour of implicit_base=True?

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/2530